src/main/java/net/dries007/tfc/common/blockentities/SheetPileBlockEntity.java

```diff
@@ -33,37 +32,34 @@ public SheetPileBlockEntity(BlockPos pos, BlockState state)
         Arrays.fill(stacks, ItemStack.EMPTY);
     }
 
-    public void addSheet(Direction direction, ItemStack stack)
+    public void addSheet(int index, ItemStack stack)
```

Comment:
I don't like exposing an `int index` here without any context to what it means. The sheet pile should still have the same interface as before: take a direction, add the sheet for that direction. It should be able to infer the index based on the rotation.

Also, this would mean this doesn't have to be a binary incompatible change, which is always nice.

Reply:
In order to apply the rotation and mirroring on the block to convert the Direction to the index (as I'm doing in `faceToIndex`), the method would need access to the BlockState.

One option would be to change `int faceToIndex(BlockState, Direction)` to something like `Direction applyRotationAndMirror(BlockState, Direction)`, then input the result to `addSheet`. Personally, I think that's even more confusing and doesn't signify to the caller that this functionality has been changed.

Another option is to change the parameter to `int stackIndex` and rename `faceToIndex()` to `mapFaceToStackIndex()` to make this relationship more obvious.

Reply:
>  the method would need access to the BlockState.

It has access, it can call `getBlockState()` in the method.